routing/localchans/manager.go

```diff
@@ -87,6 +87,22 @@ func (r *Manager) UpdatePolicy(newSchema routing.ChannelPolicy,
 		// will be used to report invalid channels later on.
 		delete(unprocessedChans, info.ChannelPoint)
 
+		nonZeroInboundFees := newSchema.InboundFee.Base != 0 ||
+			newSchema.InboundFee.Rate != 0
+
+		// Prevent setting inbound fees on private channels, because
+		// this fee can't be communicated through bolt11 invoice route
+		// hints.
+		if info.AuthProof == nil && nonZeroInboundFees {
```

Comment:
Interesting!
I see you've generalized not being able to set inbound fees on unannounced channels.
In the case of a node that pays out a lot through its private channel, the peer would be able to charge inbound fees theoretically. It's the receiving through an unannounced channel that's problematic.
I see why you took this decision. I don't have a solution. But I don't like it.

Reply:
It's only temporary. Next step is to extend the bolt11 format so that inbound fees can be communicated in invoices.

Reply:
Afaik channel updates are still exchanged between direct peers for private channels. So a sender that uses its private channel should be aware of its peer inbound fee policy?

With this PR it isn't possible to set it, but I think that with bolt11 support this if statement can be removed completely.

Reply:
But then how can you charge inbound fees then? Even if you were able to set it for private channels, it wouldn't have any effect as you say for single-hop hints. And for multi-hop hints, it wouldn't work.

Reply:
@bitromortac clarified that we might be able to remove this even without bolt11 support. 

Advantage is that inbound fees would become possible for the first hop even if the channel is private. This may help endpoint users to find peers, because those peers will have a way to charge extra for a channel that is potentially relatively stagnant.

Currently LND doesn't produce multi-hop hints, so LND invoices wouldn't be unpayable if a private channel had an inbound fee set. Also because - at least via cli - positive inbound fees aren't allowed, the worst that can happen if a multi-hop hint would be crafted manually is that the sender won't benefit from the inbound discount.

Overall seems that all of this might tip into the direction of removing the limitation?

Reply:
Removed limitation. Inbound fees can now also be set on private channels. Tested as the first hop, and it works as expected.